core/statedb/statedb.go

```diff
@@ -237,6 +238,31 @@ func (s *StateDB) syncPendingNft(pendingNft map[int64]*nft.L2Nft) error {
 	return nil
 }
 
+func (s *StateDB) SyncPendingGas() error {
+	if cacheAccount, ok := s.AccountCache.Get(types.GasAccount); ok {
+		formatAccount := cacheAccount.(*types.AccountInfo)
+		for assetId, delta := range s.PendingGasMap {
+			if asset, ok := formatAccount.AssetInfo[assetId]; ok {
+				formatAccount.AssetInfo[assetId].Balance = ffmath.Add(asset.Balance, delta)
+			} else {
+				formatAccount.AssetInfo[assetId] = &types.AccountAsset{
+					Balance: delta,
+				}
+			}
+		}
+		account, err := chain.FromFormatAccountInfo(formatAccount)
+		if err != nil {
+			return err
+		}
+		err = s.redisCache.Set(context.Background(), dbcache.AccountKeyByIndex(account.AccountIndex), account)
```

Comment:
I don't think we need to set it on Redis before we make any changes.

Reply:
updated